src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AbstractAuditLog.java

```diff
@@ -118,81 +99,18 @@
-
-        logRequestBody = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_LOG_REQUEST_BODY, true);
-        resolveIndices = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_RESOLVE_INDICES, true);
-
-        ignoredAuditUsers = new ArrayList<>(settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_IGNORE_USERS, defaultIgnoredUsers));
-
-        if(ignoredAuditUsers.size() == 1 && "NONE".equals(ignoredAuditUsers.get(0))) {
-            ignoredAuditUsers.clear();
-        }
-
-        if (ignoredAuditUsers.size() > 0) {
-            log.info("Configured Users to ignore: {}", ignoredAuditUsers);
-        }
-
-        ignoredComplianceUsersForRead = new ArrayList<>(settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_READ_IGNORE_USERS, defaultIgnoredUsers));
-
-        if(ignoredComplianceUsersForRead.size() == 1 && "NONE".equals(ignoredComplianceUsersForRead.get(0))) {
-            ignoredComplianceUsersForRead.clear();
-        }
-
-        if (ignoredComplianceUsersForRead.size() > 0) {
-            log.info("Configured Users to ignore for read compliance events: {}", ignoredComplianceUsersForRead);
-        }
-
-        ignoredComplianceUsersForWrite = new ArrayList<>(settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_IGNORE_USERS, defaultIgnoredUsers));
-
-        if(ignoredComplianceUsersForWrite.size() == 1 && "NONE".equals(ignoredComplianceUsersForWrite.get(0))) {
-            ignoredComplianceUsersForWrite.clear();
-        }
-
-        if (ignoredComplianceUsersForWrite.size() > 0) {
-            log.info("Configured Users to ignore for write compliance events: {}", ignoredComplianceUsersForWrite);
-        }
-
-        ignoreAuditRequests = settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_IGNORE_REQUESTS, Collections.emptyList());
-        if (ignoreAuditRequests.size() > 0) {
-            log.info("Configured Requests to ignore: {}", ignoreAuditRequests);
-        }
-
-        this.excludeSensitiveHeaders = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_EXCLUDE_SENSITIVE_HEADERS, true);
+        this.auditConfig = AuditConfig.getConfig(settings);
+
+        log.info("Configured audit setting for auditing on rest layer : {}", auditConfig.isRestApiAuditEnabled());
+        log.info("Configured audit setting for auditing on transport layer : {}", auditConfig.isTransportApiAuditEnabled());
+        log.info("Configured audit setting to log request body : {}", auditConfig.shouldLogRequestBody());
+        log.info("Configured audit setting to resolve bulk requests : {}", auditConfig.shouldResolveBulkRequests());
+        log.info("Configured audit setting to resolve indices : {}", auditConfig.shouldResolveIndices());
+        log.info("Configured audit setting to exclude sensitive headers : {}", auditConfig.shouldExcludeSensitiveHeaders());
+        log.info("All configured audit settings : {}", auditConfig.toString());
```

Comment:
remove this line

Reply:
I would have personally preferred this to see all configured settings. 
Why are you suggesting to remove this ?

Reply:
Is it redundant?